internal/mesh/internal/controllers/sidecarproxy/controller.go

```diff
@@ -121,13 +121,20 @@ func (r *reconciler) Reconcile(ctx context.Context, rt controller.Runtime, req c
 		rt.Logger.Error("error reading the associated workload", "error", err)
 		return err
 	}
+
 	if workload == nil {
 		// If workload has been deleted, then return as ProxyStateTemplate should be cleaned up
 		// by the garbage collector because of the owner reference.
 		rt.Logger.Trace("workload doesn't exist; skipping reconciliation", "workload", workloadID)
 		return nil
 	}
 
+	// If the workload is for a xGateway, then do nothing + let the gatewayproxy controller reconcile it
+	if gatewayKind := workload.Metadata["gateway-kind"]; gatewayKind != "" {
```

Comment:
can this panic if the key isn't present? should we use a comma ok here?

Reply:
This takes advantage of the fact that the zero-value is returned if not present in the map

https://go.dev/blog/maps#exploiting-zero-values

I think this has been a tripping hazard for reviewers across a couple PRs now, so I don't mind adding the `ok` to make things clearer